File: JSPatchProject/ViewController.m
//
//  ViewController.m
//  JSPatchProject
//
//  Created by zhcpeng on 15/12/1.
//  Copyright © 2015年 Beijing HuiMai Online Network Technology Co., Ltd. All rights reserved.
//

#import "ViewController.h"

#import "JPEngine.h"

#import "HMTestDelegate.h"

@interface ViewController ()<HMDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [JPEngine startEngine];
    NSString *sourcePath = [[NSBundle mainBundle] pathForResource:@"viewcontroller" ofType:@"js"];
    NSString *script = [NSString stringWithContentsOfFile:sourcePath encoding:NSUTF8StringEncoding error:nil];
    [JPEngine evaluateScript:script];
    
    
}

-(IBAction)btnClick:(UIButton *)button{
    NSLog(@"输出1：本地原生方法");
//    [self callFunction];
}

- (IBAction)btnAlieat:(id)sender {
//    UIAlertView *alert = [[UIAlertView alloc]initWithTitle:@"111" message:@"22" delegate:self cancelButtonTitle:@"NO" otherButtonTitles:nil, nil];
//    [alert show];
    
    HMTestDelegate *obj = [[HMTestDelegate alloc]init];
    obj.delegate = self;
    [obj callDelegate];
    
}

-(void)testDelegate{
    NSLog(@"输出4：本地代理方法");
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: JSPatchProject/viewcontroller.js
defineClass('ViewController' , {
            
    btnClick: function(button){
        console.log("输出2:JS取代原生代码方法")
        self.callFunction();
    },
            
    callFunction: function(){
        console.log("输出3:JS调用JS方法")
    },
            
    //原本类里必须引入代理后才能用JS调用，此时原生类里代理可以不用实现
    testDelegate : function(){
        console.log("输出5:JS取代原生代理方法")
    }

})


//            var alertView = require('UIAlertView').alloc().initWithTitle_message_delegate_cancelButtonTitle_otherButtonTitles("Alert","JS Alert", self, "OK", null);
//            alertView.show()

//            alertView_willDismissWithButtonIndex: function(alertView, idx) {
//            console.log('click btn ' + alertView.buttonTitleAtIndex(idx).toJS())
//            },
